GEOSwift/GEOS/GeometryConvertible+GEOS.swift

```diff
@@ -249,20 +261,49 @@ public extension GeometryConvertible {
         return try performUnaryTopologyOperation(GEOSGetCentroid_r)
     }
 
+    func minimumBoundingCircle() throws -> Circle {
+        let context = try GEOSContext()
+        let geosObject = try geometry.geosObject(with: context)
+        var radius: Double = 0
+        var optionalCenterPointer: OpaquePointer?
+        guard let geometryPointer = GEOSMinimumBoundingCircle_r(
+            context.handle, geosObject.pointer, &radius, &optionalCenterPointer) else {
+                // if we somehow end up with a non-null center and a null geometry,
+                // we must still destroy the center before throwing an error
+                if let centerPointer = optionalCenterPointer {
+                    GEOSGeom_destroy_r(context.handle, centerPointer)
+                }
+                throw GEOSError.libraryError(errorMessages: context.errors)
+        }
+        // For our purposes, we only care about the center and radius.
+        GEOSGeom_destroy_r(context.handle, geometryPointer)
+        guard let centerPointer = optionalCenterPointer else {
+            throw GEOSError.noMinimumBoundingCircle
+        }
+        let center = try Point(geosObject: GEOSObject(context: context, pointer: centerPointer))
+        return Circle(center: center, radius: radius)
+    }
+
     func polygonize() throws -> GeometryCollection {
         return try [self].polygonize()
     }
 
     // MARK: - Buffer Functions
 
-    func buffer(by width: Double) throws -> Geometry {
+    func buffer(by width: Double) throws -> Geometry? {
         let context = try GEOSContext()
         let geosObject = try geometry.geosObject(with: context)
         // returns nil on exception
         guard let resultPointer = GEOSBuffer_r(context.handle, geosObject.pointer, width, 0) else {
             throw GEOSError.libraryError(errorMessages: context.errors)
         }
-        return try Geometry(geosObject: GEOSObject(context: context, pointer: resultPointer))
+        do {
+            return try Geometry(geosObject: GEOSObject(context: context, pointer: resultPointer))
+        } catch GEOSwiftError.tooFewPoints {
+            return nil
```

Comment:
Rather than returning nil here (which occurs when the result is an empty geometry due to a negative width), let's reject negative widths upfront with a new error.